tests/emukit/quadrature/test_rbf_quadrature_kernel.py

```diff
@@ -4,25 +4,87 @@
 
 import numpy as np
 import GPy
+import pytest
 
 from emukit.model_wrappers.gpy_quadrature_wrappers import RBFGPy
-from emukit.quadrature.kernels import QuadratureRBF
+from emukit.quadrature.kernels import QuadratureRBFnoMeasure, QuadratureRBFIsoGaussMeasure
+from emukit.quadrature.kernels.integration_measures import IsotropicGaussianMeasure
 
 
-def test_rbf_qkernel_shapes():
+@pytest.fixture
+def qrbf_no_measure():
     x1 = np.array([[-1, 1], [0, 0], [-2, 0.1]])
     x2 = np.array([[-1, 1], [0, 0], [-2, 0.1], [-3, 3]])
     M1 = x1.shape[0]
     M2 = x2.shape[0]
     D = x1.shape[1]
 
     # integral bounds
-    lb = -3
-    ub = 3
+    bounds = [(-1, 2), (-3, 3)]
 
     gpy_kernel = GPy.kern.RBF(input_dim=D)
     emukit_rbf = RBFGPy(gpy_kernel)
-    emukit_qrbf = QuadratureRBF(emukit_rbf, integral_bounds=D * [(lb, ub)])
+    emukit_qrbf = QuadratureRBFnoMeasure(emukit_rbf, integral_bounds=bounds)
+    return emukit_qrbf, x1, x2, M1, M2, D
+
+
+@pytest.fixture
+def qrbf_iso_gauss_measure():
+    x1 = np.array([[-1, 1], [0, 0], [-2, 0.1]])
+    x2 = np.array([[-1, 1], [0, 0], [-2, 0.1], [-3, 3]])
+    M1 = x1.shape[0]
+    M2 = x2.shape[0]
+    D = x1.shape[1]
+
+    gpy_kernel = GPy.kern.RBF(input_dim=D)
+    emukit_rbf = RBFGPy(gpy_kernel)
+    measure = IsotropicGaussianMeasure(mean=np.arange(D), variance=2.)
+    emukit_qrbf = QuadratureRBFIsoGaussMeasure(rbf_kernel=emukit_rbf, measure=measure)
+    return emukit_qrbf, x1, x2, M1, M2, D
+
+
+def test_rbf_qkernel_no_measure_shapes(qrbf_no_measure):
+    emukit_qrbf, x1, x2, M1, M2, D = qrbf_no_measure
+
+    # kernel shapes
+    assert emukit_qrbf.K(x1, x2).shape == (M1, M2)
+    assert emukit_qrbf.qK(x2).shape == (1, M2)
+    assert emukit_qrbf.Kq(x1).shape == (M1, 1)
+    assert isinstance(emukit_qrbf.qKq(), float)
+
+    # gradient shapes
+    assert emukit_qrbf.dKq_dx(x1).shape == (M1, D)
+    assert emukit_qrbf.dqK_dx(x2).shape == (D, M2)
+
+
+def test_rbf_qkernel_no_measure_qK(qrbf_no_measure):
+    emukit_qrbf, _, x2, _, _, _ = qrbf_no_measure
+    # to check the integral, we check if it lies in some confidence interval.
+    # these intervals were computed as follows: the kernel emukit_qrbf.K was integrated in the first argument by
+    # simple random sampling with 1e6 samples. This was done 100 times. The intervals show mean\pm 3 std of the 100
+    # integrals obtained by sampling. There might be a very small chance the true integrals lies outside the specified
+    # intervals.
+    intervals = np.array([[3.047326312081091, 3.076162828036314],
+                          [5.114069393071916, 5.144207478013706],
+                          [0.9879234032609836, 1.0000191615928034],
+                          [0.07073863074148745, 0.07217298756057355]])
```

Comment:
I think this needs to be a bit larger, even up to 2 or 3 numbers after comma. With that precision things like numpy versions may cause the test to fail

Reply:
Not sure what you mean by "this" in the first sentence. If you mean the floats in the intervals I think already 4-5 digits are enough as the rest is statistical noise anyways. I should perhaps round them even.

Reply:
yes, the floats. 2-3 digits after comma would suffice, i think

Reply:
I did not round because it is very tedious to decide where to round. The numbers are stochastic anyways, so rounding should not affect if the test fails.